src/casts/specify-representation.rkt

```diff
@@ -1408,14 +1419,22 @@ but a static single assignment is implicitly maintained.
                    ,(map (lambda ([a : Immediate-Coercion])
                            (cons "argument" (sr-immediate-coercion a)))
                          a*))))]
-    [(Ref (app sr-immediate-coercion r) (app sr-immediate-coercion w))
+    [(Ref (app sr-immediate-coercion r) (app sr-immediate-coercion w) flag)
      (begin$
        (assign$ second-tag (op$ + (op$ %<< ZERO-IMDT COERCION-SECOND-TAG-SHIFT)
                                 COERCION-REF-SECOND-TAG))
-       (sr-alloc "ref-coercion" COERCION-MEDIATING-TAG
-                 `(("tag" . ,second-tag)
-                   ("read-coercion" . ,r)
-                   ("write-coercion" . ,w))))]
+       (if (cast-profiler?)
+           (sr-alloc "ref-coercion" COERCION-MEDIATING-TAG
+                     `(("tag" . ,second-tag)
+                       ("read-coercion" . ,r)
+                       ("write-coercion" . ,w)
+                       ("flag" . ,(case flag
+                                    ['GRef COERCION-REF-REF-FLAG]
+                                    ['GVect COERCION-REF-VEC-FLAG]))))
```

Comment:
I would prefer that this be:
`
(case flag
  [(GRef) ...] ...)
`
The former code works and is type safe but doesn't follow the syntax expected in case.

Reply:
Done.